### PR TITLE
Improve logging

### PR DESCRIPTION
**Pull request by johnstairs**
- Switching to [zerolog](https://github.com/rs/zerolog) for structured logging. 
- Log entries are JSON-formatted by default but this can be overridden with a command-line argument.
- Log verbosity can be set with a command-line argument.
- Each request is assigned a request ID, which is added each log message associated with the request and is returned as an HTTP response header.
- Adding a log message when the server starts and has started listening on a port

### CODE REVIEW

File: database/database.go [hunk 1]
```diff
@@ -87,7 +87,7 @@ func ConnectPostgresqlDatabase(connectionString, password string) (core.Metadata
 
 func createRepository(dialector gorm.Dialector) (core.MetadataDatabase, error) {
 	db, err := gorm.Open(dialector, &gorm.Config{
-		Logger:                 logger.Default.LogMode(logger.Warn),
+		Logger:                 gormzerolog.Logger{},
```
Comment (hansenms) [inline database/database.go:90]: Seems like a lot of whitespace here?

**Review by hansenms — APPROVED** [review]
Looks good. My only issue was that in the devcontainer the `vscode` user does not have write permissions in `/go` so build fails due to permission issues. A `sudo chown -R vscode:vscode /go` does the trick. I also see there is a golang group, but it is also not group writable in that folder, so maybe a tweak to that would work.

File: database/database.go (hunk 1 above)
Comment (johnstairs) [reply database/database.go:90]: It's what gofmt wants because of the longer field name below.

**Comment by johnstairs** [comment]
> Looks good. My only issue was that in the devcontainer the `vscode` user does not have write permissions in `/go` so build fails due to permission issues. A `sudo chown -R vscode:vscode /go` does the trick. I also see there is a golang group, but it is also not group writable in that folder, so maybe a tweak to that would work.

Will fix that in a separate PR.

**Pull request merged**